ruma-client-api/src/r0/sync/sync_events.rs

```diff
@@ -130,17 +130,21 @@ pub enum Filter {
 #[derive(Clone, Debug, Default, Deserialize, Serialize)]
 pub struct Rooms {
     /// The rooms that the user has left or been banned from.
+    #[serde(skip_serializing_if = "BTreeMap::is_empty")]
     pub leave: BTreeMap<RoomId, LeftRoom>,
 
     /// The rooms that the user has joined.
+    #[serde(skip_serializing_if = "BTreeMap::is_empty")]
     pub join: BTreeMap<RoomId, JoinedRoom>,
 
     /// The rooms that the user has been invited to.
+    #[serde(skip_serializing_if = "BTreeMap::is_empty")]
```

Comment:
Everything that's not `Option` but has `serde(skip_serializing_if)` also needs `serde(default)`. Actually that not being here already is a bug – it means we currently fail to parse a `Rooms` with any of these fields missing.